src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -119,7 +120,7 @@ private ResponseEntity<Object> createOverseasEntity(Transaction transaction,
 
         // Update company name set on the transaction and add a link to our newly created Overseas Entity
         // submission (aka resource) to the transaction (and potentially also a link for the 'resume' journey)
-        updateTransactionWithLinksAndCompanyName(transaction, overseasEntitySubmissionDto.getEntityName(), submissionId,
+        updateTransactionWithLinksAndCompanyName(transaction, overseasEntitySubmissionDto.getEntityName().getName(), submissionId,
```

Comment:
With Postman, can you create an Overseas Entity with no name block? Not saying this code is wrong but I'm interested in the answer and it might inform decisions taken above and below in this PR.

Reply:
Updated to set to null first and then update the value if we have the block.